commands/client_daemon_test.go

```diff
@@ -196,3 +204,39 @@ func TestVoucherPersistenceAndPayments(t *testing.T) {
 	assert.Contains(result, "0\t480000\t20")
 	assert.Contains(result, "0\t720000\t30")
 }
+
+func TestSelfDialStorageGoodError(t *testing.T) {
+	tf.IntegrationTest(t)
+
+	assert := assert.New(t)
+	require := require.New(t)
+
+	ctx := context.Background()
+
+	ctx, env := fastesting.NewTestEnvironment(ctx, t, fast.EnvironmentOpts{})
+	// Teardown after test ends.
+	defer func() {
+		err := env.Teardown(ctx)
+		require.NoError(err)
+	}()
+	require.NoError(env.GenesisMiner.MiningStart(ctx))
+
+	// Start mining.
+	miningNode := env.RequireNewNodeWithFunds(1000)
+	pledge := uint64(10)
+	collateral := big.NewInt(int64(1))
+	price := big.NewFloat(float64(0.001))
+	expiry := big.NewInt(int64(500))
+	fmt.Printf("creatin miner\n")
+	ask, err := series.CreateMinerWithAsk(ctx, miningNode, pledge, collateral, price, expiry)
```

Comment:
`CreateMinerWithAsk` is a bit outdated. The idea is that any series should be written such that it can be used in both a synchronous (`mining once`) and asynchronous (`mining start`). Currently `CreateMinerWithAsk` starts mining, and it shouldn't do this. Anywhere that a block is required to be mining should have a call to `CtxMiningOnce`. This might be a little more difficult at the moment as `mining start` is what registers all the protocols, so it may not even be possible at the moment. There was talk of have a `mining enable` or similar which would setup everything up, but not schedule the miner itself.